viz/Apex-Radar/src/common.ts

```diff
@@ -0,0 +1,312 @@
+    vizStyle.fillColor4.value.color,
+    vizStyle.fillColor5.value.color,
+  ];
+  const axisFontInfo: FontInfo = {
+    color: vizStyle.axisFontColor.value.color,
+    size: vizStyle.axisFontSize.value,
+    family: vizStyle.axisFontFamily.value,
+  };
+  const chart = {
+    type: 'radar',
+    toolbar: {
+      show: false,
+    },
+  };
+  const tooltip = {
+    marker: {
+      show: false,
+    },
+  };
+  return {
+    tooltip: tooltip,
+    chart: chart,
+    fill: populateFill(
+      vizStyle.fillRadar.value,
+      vizStyle.fillOpacity.value,
+      lineColors
+    ),
+    markers: populateMarkers(
+      vizStyle.showMarkers.value,
+      vizStyle.markerType.value,
+      lineColors
+    ),
+    datalabels: populateDataLabels(
+      vizStyle.showMarkers.value,
+      vizStyle.markerType.value,
+      lineColors
+    ),
+    plotoptions: populatePlotOptions(
+      vizStyle.plotColor1.value.color,
+      vizStyle.plotColor2.value.color
+    ),
+    xaxis: populateXAxis(vizStyle.enableXAxis.value, axisFontInfo, numDims),
+    yaxis: populateYAxis(vizStyle.enableYAxis.value, axisFontInfo),
+    stroke: populateStroke(lineColors),
+    legend: populateLegend(lineColors),
+  };
+}
+/**
+ * Takes fillRadar bool, fillOpacity level and lineColors to generate ApexFill
+ * @param fillRadar
+ * @param fillOpacity
+ * @param lineColors
+ */
+export function populateFill(
+  fillRadar: boolean,
+  fillOpacity: number,
+  lineColors: string[]
+) {
+  const fill: ApexFill = {
+    opacity: fillRadar ? fillOpacity : 0,
+    colors: lineColors,
+  };
+  return fill;
+}
+/**
+ * Takes showMarkers bool, the markerType and lineColors to populate ApexDataLabels
+ * @param showMarkers
+ * @param markerType
+ * @param lineColors
+ */
+export function populateMarkers(
+  showMarkers: boolean,
+  markerType: string,
+  lineColors: string[]
+) {
+  const size = showMarkers == true && markerType === 'default' ? 4 : 0;
+  const markers: ApexMarkers = {
+    size: size,
+    colors: lineColors,
+  };
+  return markers;
+}
+/**
+ * Takes showMarkers bool, the markerType and lineColors to populate ApexDataLabels
+ * @param showMarkers
+ * @param markerType
+ * @param lineColors
+ */
+export function populateDataLabels(
+  showMarkers: boolean,
+  markerType: string,
+  lineColors: string[]
+) {
+  if (showMarkers == true && markerType === 'data') {
+    const dataLabels: ApexDataLabels = {
+      enabled: true,
+      style: {
+        colors: lineColors,
+      },
+      background: {
+        enabled: true,
+        borderRadius: 2,
+        dropShadow: {
+          enabled: false,
+        },
+      },
+    };
+    return dataLabels;
+  }
+  const dataLabels: ApexDataLabels = {
+    enabled: false,
+  };
+  return dataLabels;
+}
+/**
+ * Takes 2 colors and populates ApexPlotOptions
+ * @param color1
+ * @param color2
+ */
+export function populatePlotOptions(color1: string, color2: string) {
+  const plotOptions: ApexPlotOptions = {
```

Comment:
Nit: Avoid declaring types if they can be inferred. You have type declarations in quite a few places, I'll just make one comment here.

```suggestion
  const plotOptions = {
```

If you have them included on purpose, feel free to keep them.

Reply:
Another option is to declare the return type of the function:

`export function populatePlotOptions(...): ApexPlotOptions`.

I think that is usually more useful to someone trying to understand the API.